### PR TITLE
Implement theming support for color tokens

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #70 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Implement custom tier 3 tokens to replace color tokens
* Update dependencies

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { FlatCompat } from "@eslint/eslintrc";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const compat = new FlatCompat({
+    baseDirectory: __dirname,
+    recommendedConfig: js.configs.recommended,
+    allConfig: js.configs.all
+});
+export default [...compat.extends("@aurodesignsystem/eslint-config")];
```
Comment (blackfalcon) [inline eslint.config.mjs:13]: I see what is happening here, but have these changed been updated in the Generator yet?

**Review by blackfalcon — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     "node": "^18.x || ^20.x"
   },
   "dependencies": {
-    "@aurodesignsystem/auro-badge": "^2.4.10",
+    "@aurodesignsystem/auro-badge": "^2.5.0",
```
Comment (jordanjones243) [inline package.json:23]: The work has not been done yet, but it looks like we are going to have to wait for the badge BETA release before this work moves forward.

File: src/color-segment.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  &:after {
+    border-top-color: var(--ds-auro-flightline-destination-line-color);
+  }
+}
+
+.iata {
+  color: var(--ds-auro-flight-segment-iata-text-color);
+}
+
+auro-badge {
```
Comment (jordanjones243) [inline src/color-segment.scss:24]: We should be using the versioned `auro-badge` and do the work to badge that adds the attribute to the component when we don't use it's original tag name.

**Pull request merged**